hypothesis-python/tests/nocover/test_recursive.py

```diff
@@ -132,7 +132,7 @@ def test_drawing_from_recursive_strategy_is_thread_safe():
 
     errors = []
 
-    @settings(database=None)
+    @settings(database=None, deadline=None)
```

Comment:
I'm guessing that this is just slow due to thread contention, so the generic perf advice for `st.recursive()` is probably sufficient: let's add a `max_leaves=20` to the recursive call.  If that doesn't work, I'd be surprised but just turn down the lists to `max_size=2`.

If it's still borderline (check with `--hypothesis-show-statistics`), then we can just disable the healthcheck - I don't actually want to test perf here, but if we can have fast tests we might as well 🙂

Reply:
Even with `max_leaves=20` and `max_size=2` I get:

> hypothesis.errors.FailedHealthCheck: Data generation is extremely slow: Only produced 7 valid examples in 1.37 seconds (0 invalid ones and 1 exceeded maximum size).

even locally, on a reasonably recent laptop CPU ([AMD Ryzen 7 PRO 4750U](https://www.amd.com/en/products/apu/amd-ryzen-7-pro-4750u)).

Thus, I ended up disabling the health check entirely now.